.github/workflows/scripts/generate-weekly-report.js

```diff
@@ -0,0 +1,428 @@
+    "- `{CATEGORY}: {COUNT} ({CHANGE_FROM_PREVIOUS_WEEK})`",
+    "## Issues Report",
+    '<ul>'];
+
+  // generate report for issues
+  for (const lbl of Object.keys(issuesData)) {
+    const section = [``];
+    const { count, data, title } = issuesData[lbl];
+
+    if (previousReport === null) {
+      section.push(`<li> ${title}: ${count}`);
+    } else {
+      const previousCount = previousReport.issuesData[lbl].count; 
+      section.push(`<li> ${title}: ${count} (${count - previousCount})`);
+    }
+
+    // generate summary if issues exist
+    // NOTE: the newline after <summary> is required for markdown to render correctly
+    if (data.length !== 0) {
+      section.push(`<details>
+    <summary> Issues </summary>\n`);
+    section.push(`${data.map((issue) => {
+        const { url, title, number } = issue;
+        return `- [ ] ${title} ([#${number}](${url}))`;
+      }).join('\n')}
+</details>`);
+    }
+
+    section.push('</li>');
+    out.push(section.join('\n'));
+  }
+  out.push('</ul>');
+
+  // generate report for components
+  out.push('\n## Components Report', '<ul>');
+  for (const lbl of Object.keys(componentData)) {
+    const section = [``];
+    const data = componentData[lbl];
+    const count = Object.keys(data).length;
+
+    section.push(`<li> ${lbl}: ${count}`);
+    if (data.length !== 0) {
+    // NOTE: the newline after <summary> is required for markdown to render correctly
+      section.push(`<details>
+    <summary> Components </summary>\n`);
+      section.push(`${Object.keys(data).map((compName) => {
+        const {stability} = data[compName]
+        return `- [ ] ${compName}: ${JSON.stringify(stability)}`
+      }).join('\n')}
+</details>`);
+    }
+    section.push('</li>');
+    out.push(section.join('\n'));
+  }
+  out.push('</ul>');
+
+  // add json data
+  out.push('\n ## JSON Data');
+  out.push('<!-- MACHINE GENERATED: DO NOT EDIT -->');
+  out.push(`<details>
+<summary>Expand</summary>
+<pre>
+{
+  "issuesData": ${JSON.stringify(issuesData, null, 2)},
+  "componentData": ${JSON.stringify(componentData, null, 2)}
+}
+</pre>
+</details>`);
+  const report = out.join('\n');
+  return report;
+}
+
+async function createIssue({ github, lookbackData, report, context }) {
+  const title = `Weekly Report: ${lookbackData.sevenDaysAgo.toISOString().slice(0, 10)} - ${lookbackData.midnightYesterday.toISOString().slice(0, 10)}`;
+  return github.rest.issues.create({
+    owner: context.payload.repository.owner.login,
+    repo: "opentelemetry-collector-contrib",
+    title,
+    body: report,
+    labels: ["report"]
+  })
+}
+
+async function getLastWeeksReport({ github, since, context }) {
+  const issues = await github.rest.issues.listForRepo({
+    owner: context.payload.repository.owner.login,
+    repo: 'opentelemetry-collector-contrib',
+    state: 'all', // To get both open and closed issues
+    labels: ["report"],
+    since: since.toISOString(),
+    per_page: 1,
+    direction: "asc"
```

Comment:
I propose to also explicitly specify sort field ("created" is the [default](https://docs.github.com/en/free-pro-team@latest/rest/issues/issues?apiVersion=2022-11-28#list-repository-issues)), as having only direction specified looks a bit weird to my taste.

```suggestion
    sort: "created",
    direction: "asc"
```